cms/datavis/blocks/charts.py

```diff
@@ -247,6 +267,38 @@ def validate_x_axis(self, value: "StructValue") -> None:
                 }
             )
 
+    def validate_y_axis(self, value: "StructValue") -> None:
+        if value.get("y_axis").get("custom_reference_line"):
+            if value.get("select_chart_type") == BarColumnChartTypeChoices.BAR:
+                raise blocks.StructBlockValidationError(
+                    block_errors={
+                        "y_axis": blocks.StructBlockValidationError(
+                            block_errors={
+                                "custom_reference_line": ValidationError(
+                                    "Custom reference line is not supported for bar charts.",
+                                    code=self.ERROR_HORIZONTAL_BAR_NO_CUSTOM_REFERENCE_LINE,
+                                )
+                            }
+                        )
+                    }
+                )
+
+            if len(value["table"].headers) > self.MAX_SERIES_COUNT_WITH_CUSTOM_REFERENCE_LINE + 1:
+                # NOTE Currently this prevents use of custom reference line even
+                # when the second series is a line overlay.
+                raise blocks.StructBlockValidationError(
+                    {
+                        "y_axis": blocks.StructBlockValidationError(
+                            block_errors={
+                                "custom_reference_line": ValidationError(
+                                    "Custom reference line is supported for single-series data only.",
+                                    code=self.ERROR_MULTIPLE_SERIES_NO_REFERENCE_LINE,
+                                )
+                            }
+                        )
+                    }
+                )
+
```

Comment:
question: is there a need to validate whether the reference line is within the data bounds? Or can it be anywhere?